firehose/firehose_kinesis.go

```diff
@@ -23,23 +24,25 @@ type FirehoseWriter struct {
 type FirehoseWriterConfig struct {
 	// The value of this field is used as the firehose stream name
 	StreamName string
-	// AWS region the stream lives in
-	Region string `toml:"region"`
-	// Interval at which accumulated messages should be bulk put to
-	// firehose, in milliseconds (default 1000, i.e. 1 second).
-	FlushInterval uint32 `toml:"flush_interval"`
+	// AWS region the firehose stream lives in
+	Region string
+	// Interval at which accumulated messages should be bulk put to firehose
+	FlushInterval time.Duration
 	// Number of messages that triggers a push to firehose
-	// (default to 1, maximum is 500)
-	FlushCount int `toml:"flush_count"`
+	// Max is 500, see: http://docs.aws.amazon.com/firehose/latest/dev/limits.html
+	FlushCount int
 	// Size of batch that triggers a push to firehose
-	// (default to 1024 * 1024 (1mb))
-	FlushSize int `toml:"flush_size"`
+	// Max is 4Mb (4*1024*1024), see: http://docs.aws.amazon.com/firehose/latest/dev/limits.html
```

Comment:
5Mb is the total firehose rate limit (default -- this can be increased), whereas 4Mb is the limit per batch.

---

- By default, each Firehose delivery stream can accept a maximum of 2,000 transactions/second, 5,000 records/second, and 5 MB/second
- The PutRecordBatch operation can take up to 500 records per call or 4 MB per call, whichever is smaller.